lighthouse-core/report/handlebar-helpers.js

```diff
@@ -0,0 +1,233 @@
+      return options.fn(this);
+    } else {
+      // eslint-disable-next-line no-invalid-this
+      return options.inverse(this);
+    }
+  }
+
+  // value !== value2
+  static ifNotEq(lhs, rhs, options) {
+    if (lhs !== rhs) {
+      // eslint-disable-next-line no-invalid-this
+      return options.fn(this);
+    } else {
+      // eslint-disable-next-line no-invalid-this
+      return options.inverse(this);
+    }
+  }
+
+  // arg1 && arg2 && ... && argn
+  static and(...args) {
+    let arg = false;
+    for (let i = 0, n = args.length - 1; i < n; i++) {
+      arg = args[i];
+      if (!arg) {
+        break;
+      }
+    }
+    return arg;
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  static sanitize(str, opts) {
+    // const isViewer = opts.data.root.reportContext === 'viewer';
+
+    // Allow the report to inject HTML, but sanitize it first.
+    // Viewer in particular, allows user's to upload JSON. To mitigate against
+    // XSS, define a renderer that only transforms links and code snippets.
+    // All other markdown ad HTML is ignored.
+    const renderer = new marked.Renderer();
+    renderer.em = str => `<em>${str}</em>`;
+    renderer.link = (href, title, text) => {
+      title = title || text;
+      return `<a href="${href}" target="_blank" rel="noopener" title="${title}">${text}</a>`;
+    };
+    renderer.codespan = function(str) {
+      return `<code>${str}</code>`;
+    };
+    // eslint-disable-next-line no-unused-vars
+    renderer.code = function(code, language) {
+      return `<pre>${code}</pre>`;
+    };
+    // Nuke wrapper <p> tag that gets generated.
+    renderer.paragraph = function(str) {
+      return str;
+    };
+
+    try {
+      str = marked(str, {renderer, sanitize: true});
+    } catch (e) {
+      // Ignore fatal errors from marked js.
+    }
+
+    // The input str has been sanitized and transformed. Mark it as safe so
+    // handlebars renders the text as HTML.
+    return new Handlebars.SafeString(str);
+  }
+
+  // format time
+  static formatTime(date) {
+    const options = {
+      day: 'numeric', month: 'numeric', year: 'numeric',
+      hour: 'numeric', minute: 'numeric', second: 'numeric',
+      timeZoneName: 'short'
+    };
+    let formatter = new Intl.DateTimeFormat('en-US', options);
+
+    // Force UTC if runtime timezone could not be detected.
+    // See https://github.com/GoogleChrome/lighthouse/issues/1056
+    const tz = formatter.resolvedOptions().timeZone;
+    if (!tz || tz.toLowerCase() === 'etc/unknown') {
+      options.timeZone = 'UTC';
+      formatter = new Intl.DateTimeFormat('en-US', options);
+    }
+    return formatter.format(new Date(date));
+  }
+
+  static getHelpers() {
+    return {
+      nameToLink(name) {
+        return HandlebarHelpers.nameToLink(name);
+      },
+
+      getItemRating(value) {
+        return HandlebarHelpers.getItemRating(value);
+      },
+
+      'getTotalScore': HandlebarHelpers.getTotalScore,
```

Comment:
> We can probably structure all of these helpers like this without having to create new functions

maybe we should just avoid defining this as a `class` and have it be an object literal? That way in `report-generator.js` it could just be

```js
const handlebarHelpers = require('./handlebar-helpers');
// ...
Handlebars.registerHelper(handlebarHelpers);
```